controllers/cluster_controller_phases.go

```diff
@@ -49,8 +49,8 @@ func (r *ClusterReconciler) reconcilePhase(_ context.Context, cluster *clusterv1
 		cluster.Status.SetTypedPhase(clusterv1.ClusterPhaseProvisioning)
 	}
 
-	// Set the phase to "provisioned" if the infrastructure is ready and APIEndpoints exists
-	if cluster.Status.InfrastructureReady && len(cluster.Status.APIEndpoints) > 0 {
+	// Set the phase to "provisioned" if the infrastructure is ready and ControlPlaneEndpoint is not zero.
```

Comment:
I'm not sure we need this comment any more 😛

Reply:
Fixed!